### ISSUE TITLE
No matching formats

### ISSUE THREAD

**Issue by rodrigorega** [meeb/tubesync#336]
Type: bug/misconfig ????

### The issue

[With this video](https://www.youtube.com/watch?v=ZSxR3gy6DRc) from [this source](https://www.youtube.com/c/@PhatDiggaz) I got no download:

>Media cannot be downloaded because it has no formats which match the source requirements.

This is the config for the source:

```
Desired format	4320p (video:VP9, audio:OPUS) 60FPS
Fallback	Get next best resolution or codec instead
Downloaded    X
Can download  X	
Available formats	
ID: 91 , (256x144), fps:15.0, video:avc1.42c00b @290.288k , audio:mp4a.40.5 @0.0k / Hz
ID: 92 , (426x240), fps:30.0, video:avc1.4d4015 @546.239k , audio:mp4a.40.5 @0.0k / Hz
ID: 93 , (640x360), fps:30.0, video:avc1.4d401e @1209.862k , audio:mp4a.40.2 @0.0k / Hz
ID: 94 , (854x480), fps:30.0, video:avc1.4d401f @1568.726k , audio:mp4a.40.2 @0.0k / Hz
ID: 95 , (1280x720), fps:30.0, video:avc1.4d401f @2969.452k , audio:mp4a.40.2 @0.0k / Hz
ID: 96 , (1920x1080), fps:30.0, video:avc1.640028 @5420.722k , audio:mp4a.40.2 @0.0k / Hz
Matched formats	Combined: no match
Audio: no match
Video: no match
```
### Expected result

Video is downloaded. My goal is to download the best quality available for this source. I do not know if I'm missing something. This is the newest video in this source. All pervious 69 videos where downloaded as expected.

### Software versions

- TubeSync version 0.12.0
- yt-dlp version 2023.01.06
- FFmpeg version N-109474-gc94988a781-20230103

 Do you have any ideas?. Thanks in advance.

**Comment by bee-san** [comment]
I believe this is because `yt-dlp` updated (youtube broke how channel IDs work) see https://github.com/yt-dlp/yt-dlp/issues/6247

TL;DR update your yt-dlp

**Comment by meeb** [comment]
@bee-san thanks for trying to assist, however this looks like a failure of the matching logic which is purely in tubesync. I need to add a test and check where the matching is failing.

@rodrigorega likely this is because the only "matching" streams are the complete opposite of your source requirements (you want 8k vp9 with opus in 60fps, the only match is 1080p h264 m4a in 30fps) and you've found an edge case where no match is detected even with fallbacks. This should find a match to something, but looks like it isn't in your case.

**Comment by bee-san** [comment]
@meeb  In an attempt to be helpful, I am getting the same error on all recently added sources even with the updated yt-dlp 🙈 

I have set my quality to this, which I believe should fallback to any quality at all:

<img width="535" alt="Screenshot 2023-02-20 at 07 28 50" src="https://user-images.githubusercontent.com/10378052/220039964-9cfa49b9-b554-41be-b05e-ba81eadc5b52.png">

I get this error:

>  Media cannot be downloaded because it has no formats which match the source requirements.

I have updated tube-sync to `latest` and still get it.

It looks like my sources which I added a few weeks ago still work, it is only newer sources :)

**Comment by meeb** [comment]
@bee-san can you post an example video that isn't matching anything and your source requirements for that media item? Thanks!

**Comment by bee-san** [comment]
Sure! I added 3 sources which fail on all videos. Here's 1 example from each:

# Leah's Field Notes

<img width="553" alt="Screenshot 2023-02-20 at 07 52 22" src="https://user-images.githubusercontent.com/10378052/220045029-c7aae04d-4d57-4940-aa04-0eef50500a40.png">
<img width="687" alt="Screenshot 2023-02-20 at 07 52 57" src="https://user-images.githubusercontent.com/10378052/220045145-04b1c799-0d2a-4eeb-9ee3-bfdb8957ce56.png">
https://www.youtube.com/watch?v=UxVwEXxNvtU

# Patrick Boyle
<img width="705" alt="Screenshot 2023-02-20 at 07 53 44" src="https://user-images.githubusercontent.com/10378052/220045323-0b685095-5e9f-439b-9344-0351eecf626d.png">
<img width="666" alt="Screenshot 2023-02-20 at 07 54 11" src="https://user-images.githubusercontent.com/10378052/220045422-20c82b32-9ffe-4931-a3f5-8794571dbead.png">
https://www.youtube.com/watch?v=x-n1MVbIPLc

# How Money Works
<img width="728" alt="Screenshot 2023-02-20 at 07 54 35" src="https://user-images.githubusercontent.com/10378052/220045517-8398abf0-d00f-49cd-943c-b8abcc604870.png">
<img width="707" alt="Screenshot 2023-02-20 at 07 54 50" src="https://user-images.githubusercontent.com/10378052/220045577-0d3cdc40-0269-49cf-8d73-8801790cbfc3.png">
https://www.youtube.com/watch?v=3VqjOgPHVdg

**Comment by meeb** [comment]
@bee-san that's an issue with `yt-dlp` obtaining metadata, there is no metadata to match not that there is metadata which isn't match-able (the original post on this issue). This is fixed in the latest tubesync which has an updated `yt-dlp` bundled with it. Please pull the `:latest` docker image and restart it and this should be fixed for you.

**Comment by rodrigorega** [comment]
I pulled the latest docker image. The source has a new video and it downloaded correctly. The previous video that failed is still with the message "no matching formats". Is this expected? How I can force to download it?

**Comment by meeb** [comment]
@rodrigorega try skipping then unskipping the video, that should refresh the metadata and tasks for that media item if it has been marked as permanently failed.

**Comment by bee-san** [comment]
Just to confirm and end up my mini-issue-derailment, I ran `docker compose down`, deleted my docker images and ran `docker compose up` (docker compose pull would also work, but I wanted to delete a bunch of images anyway).

I then deleted my sources and re-added them, and I can confirm this works!

**Comment by rodrigorega** [comment]
> @rodrigorega try skipping then unskipping the video, that should refresh the metadata and tasks for that media item if it has been marked as permanently failed.

This worked. Thanks for your help.

**Comment by meeb** [comment]
@rodrigorega do you have any remaining issues or can I close this?

**Comment by rodrigorega** [comment]
All done from my side. Thak you all!
